CONTRIBUTING.md

```diff
@@ -298,6 +304,154 @@ OS and is built by Android NDK. This often introduces compatibility issues as
 Android (specifically Termux) is not a standard platform. Do not expect there
 are exist package recipes available out-of-box.
 
+## Commit guidelines
+
+A commit message should look something like:
```

Comment:
Most important should be that the commit messages are logical, i.e. states what is done and what packages are affected. 

Since not all contributors agree that this style is preferable I suggest we make the style preferable but optional. We can add an introduction paragraph here like

```
Commit messages should describe the changes done, so that maintainers can understand what was done, and to what package or scope, without having to look at the code changes. One good (but not mandatory) way to make sure the commit message fulfills these requirements is to write it on the format:
```

Reply:
To elaborate on that. A commit message should normally explain what is being changed, why it is being changed or the effect of the commit and optionally (summary of) how changes were made if the changes were logically complex. Whoever may be reading the commit messages may not understand the code itself, like the author does or at all.

Maybe a link to some resource site would be useful. Maybe something like https://who-t.blogspot.com/2009/12/on-commit-messages.html or https://wiki.openstack.org/wiki/GitCommitMessages

>Since not all contributors agree that this style is preferable I suggest we make the style preferable but optional. 

Yeah, I guess that's reasonable. But would be better to restrict it to either `package: Some description` or `change_type(package): Some description` formats, since that way there will at least be some consistency.

Reply:
> Yeah, I guess that's reasonable. But would be better to restrict it to either package: Some description or change_type(package): Some description formats, since that way there will at least be some consistency.

I totally agree on this. Although it might not seem good forcefully do so, but definitely helps when going through old commits.

Reply:
We are not forcing everyone to use specific formats for all repos, just our own termux-packages. Open source projects often have specific formats for their own repos, which need to be followed by all internal and external devs contributing to it. People can use whatever they want in their own repos. Currently, internal devs don't agree on one format and we have 2 different formats being used, which are both fine and will keep history clean, so might as well just restrict it to those 2 for all internal and external devs, instead of allowing commiters to use random formats and messing with the history.

Reply:
Done. Thanks, for all the valuable feedback. Not forcing to follow these new guidelines might be helpful as I agree that these guidelines are pretty hard to follow. I hope we may soon in future get used to these, though :)